### ISSUE TITLE
[QUEST] "Gol'Dir" in Alterac Mountains

### ISSUE THREAD

**Issue by blackcrowX** [chromiecraft/chromiecraft#477]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [ X ] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [ X ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ X ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

After proceding to acquire the Rusted Iron Key [ID: 3704] and attempting to turn in the Quest at the NPC Gol'Dir. 
https://de.classic.wowhead.com/item=3704/verrosteter-eisenschl%C3%BCssel

Gol'Dir doesn't accept the Quest Item with the same Name and same ID as acquired.
https://de.classic.wowhead.com/npc=2316/goldir

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Gol'Dir accepts the Key and the Quest is over. 
https://de.classic.wowhead.com/quest=503/goldir

After that the Quest "Blackmoore's Legacy" becomes available.
https://de.classic.wowhead.com/quest=506/blackmoores-verm%C3%A4chtnis

##### STEPS TO REPRODUCE THE PROBLEM:

1.  Accept the Quest "Gol'Dir" at the NPC "Krusk"
2.  Acquire the Item "Rusted Iron Key"
3.  Attempt to turn in quest at NPC "Gol'Dir"
4.  Be baffled by the unwillingness of the Gol'Dir to escape.

##### EXTRA NOTES:

Type here.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/91081f4ad8fed98d7bdc9c0ae13d05c86032df9f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by blackcrowX** [comment]
![Uploading WoWScrnShot_042721_223357.jpg…]()

**Comment by blackcrowX** [comment]
![WoWScrnShot_042721_223357](https://user-images.githubusercontent.com/79520609/116311943-1eed1700-a7ac-11eb-9b54-020e038eb693.jpg)
![WoWScrnShot_042721_223400](https://user-images.githubusercontent.com/79520609/116311947-20b6da80-a7ac-11eb-9529-af4800d6ce38.jpg)

**Comment by Azcobu** [comment]
Though it sounds weird, this seems to be actually Blizz-like:

"It's not broken just a little buggy. If you see a grey question mark and you have the key, just run over the entrance to the room he's held in, I had to shift out of cat form for it to recognize me." (https://classic.wowhead.com/quest=503/goldir#comments:id=2733110)

"Ok so I hear this quest is bugged -- WRONG [...] Run to him, his question mark becomes yellow as soon as you hit the entrance to his room. If the quest is still grey and you have the key rewalk into the room until it changes -- you're probably lagging." (https://tbc.wowhead.com/quest=503/goldir#comments:id=2733416)

I just tried it twice, and both times there was a grey question mark like you said:

![WoWScrnShot_042821_215755](https://user-images.githubusercontent.com/81782124/116403889-41933480-a86d-11eb-864c-935efd55b9fd.jpg)

But both times when I ran a few steps out of the room and came back inside, the question mark turned yellow and I could hand the quest in:

![WoWScrnShot_042821_215806](https://user-images.githubusercontent.com/81782124/116403926-4b1c9c80-a86d-11eb-9187-bfe12aff0f9f.jpg)

So it looks like it's behaving just as it did on the live servers. Not ideal, but not an Azeroth Core bug either.
